x/incentives/keeper/gauge.go

```diff
@@ -110,13 +123,52 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		}
 	}
 
-	// check if denom this gauge pays out to exists on-chain
-	if !k.bk.HasSupply(ctx, distrTo.Denom) && !strings.Contains(distrTo.Denom, "osmovaloper") {
-		return 0, fmt.Errorf("denom does not exist: %s", distrTo.Denom)
+	nextGaugeId := k.GetLastGaugeID(ctx) + 1
+
+	// For no lock gauges, a pool id must be set.
+	// A pool with such id must exist and be a concentrated pool.
+	if distrTo.LockQueryType == lockuptypes.NoLock {
+		if poolId == 0 {
+			return 0, fmt.Errorf("no lock gauges must have a pool id")
+		}
+
+		// If not internal gauge denom, then must be set to ""
+		// and get overwritten with the external prefix + pool id
+		// for internal query purposes.
+		if distrTo.Denom != types.NoLockInternalGaugeDenom(poolId) {
+			// If denom is set, then fails.
+			if distrTo.Denom != "" {
+				return 0, fmt.Errorf("no lock external gauges must have an empty denom set")
```

Comment:
```suggestion
				return 0, fmt.Errorf("lock external gauges must have an empty denom set")
```
might also be helpful in the error to know what was provided

Reply:
This is correct to be no lock. No lock is a type of a gauge but going to format the denom

Done here: [975275b](https://github.com/osmosis-labs/osmosis/pull/5459/commits/975275bbb4ba4841198502cf02ab3c3abc85813e)